src/frontends/tensorflow_common/src/op/matrix_band_part.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/op/logical_and.hpp"
+#include "openvino/op/less_equal.hpp"
+#include "openvino/op/unsqueeze.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/reshape.hpp"
+#include "utils.hpp"
+
+using namespace std;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+OutputVector translate_matrix_band_part_op(const NodeContext& node) {
+    // Input tensor and parameters
+    auto input = node.get_input(0);
+
+    // Get the shape of the input tensor
+    auto input_shape = make_shared<v3::ShapeOf>(input);
+
+    // Find sizes m and n of the last two dimensions
+    auto m = make_shared<v1::StridedSlice>(input_shape,
+                                           make_shared<v0::Constant>(element::i64, Shape{1}, {-2}),
+                                           make_shared<v0::Constant>(element::i64, Shape{1}, {-1}),
+                                           make_shared<v0::Constant>(element::i64, Shape{1}, {1}),
+                                           std::vector<int64_t>({0}),
+                                           std::vector<int64_t>({1}));
+
+    auto n = make_shared<v1::StridedSlice>(input_shape,
+                                           make_shared<v0::Constant>(element::i64, Shape{1}, {-1}),
+                                           make_shared<v0::Constant>(element::i64, Shape{1}, {}),
+                                           make_shared<v0::Constant>(element::i64, Shape{1}, {1}),
+                                           std::vector<int64_t>({0}),
+                                           std::vector<int64_t>({1}));
+
+    // Generate ranges [0, m) and [0, n)
+    auto range_m = make_shared<v1::Range>(m);
+    auto range_n = make_shared<v1::Range>(n);
+
+    // Unsqueeze ranges to have tensors of shapes [m, 1] and [1, n]
+    auto unsqueeze_range_m = make_shared<v0::Unsqueeze>(range_m, make_shared<v0::Constant>(element::i64, Shape{1}, {1}));
+    auto unsqueeze_range_n = make_shared<v0::Unsqueeze>(range_n, make_shared<v0::Constant>(element::i64, Shape{1}, {0}));
+
+    // Create indicator function using logical operations
+    auto in_band_rows = make_shared<v1::LogicalAnd>(make_shared<v1::LessEqual>(unsqueeze_range_m, m),
+                                                    make_shared<v1::GreaterEqual>(unsqueeze_range_m, make_shared<v0::Constant>(element::i64, Shape{}, {0})));
+
+    auto in_band_cols = make_shared<v1::LogicalAnd>(make_shared<v1::LessEqual>(unsqueeze_range_n, n),
+                                                    make_shared<v1::GreaterEqual>(unsqueeze_range_n, make_shared<v0::Constant>(element::i64, Shape{}, {0})));
+
+    auto in_band_indicator = make_shared<v1::LogicalAnd>(in_band_rows, in_band_cols);
```

Comment:
you forget to use other inputs to this operation `num_lower` and `num_upper`. `in_band` computes as follows (see https://www.tensorflow.org/api_docs/python/tf/raw_ops/MatrixBandPart):

```
in_band(m, n) = (num_lower < 0 || (m-n) <= num_lower)) && (num_upper < 0 || (n-m) <= num_upper)
```